backend/libinput/pointer.c

```diff
@@ -131,9 +136,13 @@ void handle_pointer_axis(struct libinput_event *event,
 				wlr_event.orientation = WLR_AXIS_ORIENTATION_HORIZONTAL;
 				break;
 			}
-			double hr_delta =
-					libinput_event_pointer_get_axis_value_v120(pevent, axes[i]);
-			wlr_event.delta = hr_delta * 0.125; // assuming 15 degree/detent wheel
+			if (wlr_event.source == WLR_AXIS_SOURCE_WHEEL) {
+				// send scaled hi-res value assuming 15 degree/detent wheel
+				wlr_event.delta =
+						libinput_event_pointer_get_axis_value_v120(pevent, axes[i]) * 0.125;
+			} else {
+				wlr_event.delta = libinput_event_pointer_get_axis_value(pevent, axes[i]);
+			}
```

Comment:
you can calculate the right amount of movement by comparing the v120 value with the degrees. if you get 5 degrees and a v120 of 30, you know the full click angle is 20. But that in itself is a problem for all the reasons I pointed out in that email I sent out. And which I publish as blog post soon. So for now, scaling to 15 is the best thing you can do, but I'd do this based on the click angle so you always get a value of 15 regardless of the physical click angle (instead of hardcoding 1/8).